server.js

```diff
@@ -18,15 +19,134 @@ import cors from 'cors'
+    res.status(404).json(ERROR_SHOWS_NOT_FOUND)
+  } else {
+    res.json(netflixId)
+  }
+})
+
+
+// e) shows movies and shows from a specific country 
+app.get('/shows/country/:country', (req, res) => {
+  const country = req.params.country
+  const netflixCountry = netflixData.filter((item) => item.country === country)
+
+  //this shows an error message if there are no movies or shows from a specific country
+  if (netflixCountry.length === 0) {
+    res.status(404).json(ERROR_SHOWS_NOT_FOUND)
+  } else {
+    res.json(netflixCountry)
+  }
+})
+
+
+// f) Shows movies and tv-shows from a specified genre. If you search for e.g Comedies you will also get results where many different genres are listed for the same movie/show. It also works to write comedies (lowercase c) or even only part of a word like comed.
+app.get('/shows/genre/:genre', (req, res) => {
+  const genre = req.params.genre
+  const showsGenre = netflixData.filter((item) => item.listed_in.toLowerCase().includes(genre.toLowerCase()))
+
+  //shows error message if the specified word isn't included within a genre.
+  if (showsGenre.length === 0) {
+    res.status(404).json(ERROR_SHOWS_NOT_FOUND)
+  } else { 
+    res.json(showsGenre)
+  }
+  })
+
+
+// g) Similar to the one above but here you will only get the movies/shows which is defined with only one genre, for example Comedies (but has to be written exactly like that).
+app.get('/genre/:genre', (req, res) => {
+  const genre = req.params.genre
+  const movieGenre = netflixData.filter((item) => item.listed_in === genre)
+
+//shows an error message if a genre doesn't exist or is misspelled. 
+  if (movieGenre.length === 0) {
+    res.status(404).json(ERROR_SHOWS_NOT_FOUND)
+  } else { 
+    res.json(movieGenre)
+}
 })
 
+
+// h) Here we can query a movie/show title. We can use the query method since several movies/shows can have the same name. We have to write like this in the browser: localhost:8080/shows/title?title=xxx
+app.get('/shows/title', (req, res) => {
+  const title  = req.query.title
+  if (title) {
+    const filteredTitle = netflixData.filter((item) => item.title === title)
+    res.json(filteredTitle)
+  } else {
+    res.json(netflixData)
+  }
+})
+
+
+ // i) Two params are used to return movies that are of the type "Movie" or"TV Show" and from a specified year.
+app.get('/shows/:year/type/:type', (req, res) => {
+  const { year, type } = req.params;
+  const yearType = netflixData.filter((yearType) => {
+    return +yearType.release_year === +year && yearType.type === type
+  })
+//error message that is displayed if one or two of the parameteras don't exist
+  if (yearType.length === 0) {
+    res.status(404).json(ERROR_SHOWS_NOT_FOUND)
+  } else {
+    res.json(yearType)
+  } 
+})
+
+ // j) DUMMY ENDPOINT: Create some empty/dummy endpoints which could contain more complex operations in the future.  Find good names for them. (RED LEVEL)
+ app.get('/shows/maxduration/:maxduration', (req, res) => {
```

Comment:
Let's comment this endpoint for now, since it's not doing anything. Nonetheless, good job!